File: 02 Algorithm Reference/12 Reality Modelling/05 Fill Models/01 code.cs
// Set the fill models in initialize:
Securities["IBM"].SetFillModel(new PartialFillModel());

// Custom fill model implementation stub
public class PartialFillModel : ImmediateFillModel {
    public override OrderEvent MarketFill(Security asset, MarketOrder order) {
        //Override order event handler and return partial order fills, 
    }
}

File: 02 Algorithm Reference/12 Reality Modelling/05 Fill Models/01 code.py
# Set the fill models in initialize:
self.SetFillModel(PartialFillModel(), self.Securities["IBM"])

# Custom fill model implementation stub
class PartialFillModel(ImmediateFillModel):
    def MarketFill(self, asset, order):
        # Override order event handler and return partial order fills
        pass